saleor/checkout/models.py

```diff
@@ -104,14 +106,31 @@ def get_subtotal(self, discounts=None, taxes=None):
         subtotals = (line.get_total(discounts, taxes) for line in self)
         return sum(subtotals, ZERO_TAXED_MONEY)
 
-    def get_total(self, discounts=None, taxes=None):
-        """Return the total cost of the checkout."""
+    def get_total_without_gift_cards(self, discounts=None, taxes=None):
+        """Return the total cost of the checkout without gift cards."""
         return (
             self.get_subtotal(discounts, taxes)
             + self.get_shipping_price(taxes)
             - self.discount_amount
         )
 
+    def get_total(self, discounts=None, taxes=None):
+        """Return the total cost of the checkout."""
+        total = (
+            self.get_total_without_gift_cards(discounts, taxes)
+            - self.get_total_gift_cards_balance()
+        )
+        return max(total, ZERO_TAXED_MONEY)
+
+    def get_total_gift_cards_balance(self):
+        """Return the total balance of the gift cards assigned to the checkout."""
+        gift_cards_total_balance = self.gift_cards.aggregate(
+            models.Sum("current_balance")
+        )["current_balance__sum"]
+        if gift_cards_total_balance:
```

Comment:
You could just do: `return gift_cards_total_balance or ZERO_MONEY`. I would simplify naming here as well, e.g. `gift_cards_total_balance` could be simply `balance`.